.husky/pre-commit

```diff
@@ -2,3 +2,5 @@
 . "$(dirname "$0")/_/husky.sh"
 
 npm run lint && npm run format && npm run build
+# Stage modified files
+git add $(git diff --name-only)
```

Comment:
_It might be prudent to move this change to a separate PR for more discussion on expected behavior._

Unless we limit this to all files in the `dist/` directory, it will stage and commit all changes.

**Example**
1. Create one-off branch from `refactor/typescript`
2. Make trivial change to `README.md`
3. Make trivial change to `src/main.ts`
4. Stage `README.md` and commit it.

Expected result: it should only commit the changes to the `README.md` file.
Actual result: it builds files in `dist/`, stages them, and also stages `src/main.ts` before commit.

Perhaps we should only run lint/format/build when we detect that files in `src/` have been staged for commit? That would ensure that builds only occur when files that affect the build have been modified.